Added homework TicTacToe

kshrnkrstn opened the pull request.

kshrnkrstn committed:
Added homework TicTacToe

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,190 @@
+﻿//отрисовка игрового поля------------------------------------------------------------------------------------
+void DrawField(int[] playersMove)
+{
+    Console.WriteLine("-------------");
```
Этот литерал несколько раз используется. Давай вынесем в константы метода

kshrnkrstn replied:
Исправлено

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,190 @@
+{
+    Console.WriteLine("-------------");
+    Console.Write("| "); IsEmptyCell(playersMove[0]);Console.Write(" | "); IsEmptyCell(playersMove[1]);
+    Console.Write(" | "); IsEmptyCell(playersMove[2]); Console.WriteLine(" |");
```
Литералы типа `" | "` тоже давай в константы вынесем

kshrnkrstn replied:
Исправлено

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,190 @@
+    Console.Write(" | "); IsEmptyCell(playersMove[8]); Console.WriteLine(" |");
+    Console.WriteLine("-------------\n");
+}
+//подготовка ячейки для отрисовки-----------------------------------------------------------------------------
```
Вместо комментария такого вида нужно использовать summary (через ///). Как это делал показывал на лекциях

kshrnkrstn replied:
Исправлено

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,190 @@
+//подготовка ячейки для отрисовки-----------------------------------------------------------------------------
+void IsEmptyCell(int cell)
+{
+    if (cell == 10) //ход X
```
такие комментарии не нужны

Dezodemius replied:
10 и 11 это какие-то магические числа. Давай их выделим в переменные или константы

kshrnkrstn replied:
Исправлено

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,190 @@
+{
+    
+    char turn = 'v';
+    while (turn != 'x' && turn != 'o' && turn != 'O' && turn != 'X')
```
'Х' и 'O' давай вынесем в константы и везде их будем использовать. Проверять на x или Х не нужно. Пусть всегда будем работать с Х

kshrnkrstn replied:
Исправлено

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,190 @@
+    }
+    else
+    {
+        Console.WriteLine("Ответьте корректно!");
```
Непонятно, что значит корректно. Если и выводить сообщение, то так, чтобы оно было однозначное

kshrnkrstn replied:
Исправлено

Dezodemius commented on TicTacToe/TicTacToe/TicTacToe.csproj:
```diff
@@ -0,0 +1,10 @@
+    <PropertyGroup>
+        <OutputType>Exe</OutputType>
+        <TargetFramework>net8.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
```
`ImplicitUsings` давай будем удалять из проектов

kshrnkrstn replied:
Исправлено

kshrnkrstn committed:
Deleted <ImplicitUsings> from TicTacToe.csproj

kshrnkrstn committed:
Updated functions comments

Комментарии изменены на <summary>
Ненужные комментарии удалены

kshrnkrstn committed:
added using System

kshrnkrstn committed:
Update void DrawField

kshrnkrstn committed:
Magic numbers changed to constants

kshrnkrstn committed:
Update IsWinner function

playersMove изменено на playersMoves . Что-то в коде ломалось, но что - мы решили не выяснять.

kshrnkrstn committed:
Added char constants for 'X' and 'O'

- вместо каждого прописывания X и Y созданы константы
- требование: все ответы игрока в верхнем регистре
- исправлены консольные сообщения

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,212 @@
+﻿using System;
+
```
Давай переделаем на класс и метод Main. Вместо операторов верхнего уровня

kshrnkrstn replied:
Исправлено

Dezodemius commented on TicTacToe/TicTacToe/Program.cs:
```diff
@@ -0,0 +1,212 @@
+/// </summary>
+void StartGame(int[] playersMoves)
+{
+    const char turnX = 'X';
```
А вот тут у тебя они как константы, а строчками выше всё равно как литералы. Давай везде, где используются эти символы будем использовать константы

kshrnkrstn replied:
Исправлено

kshrnkrstn committed:
Replace functions to class Program and fixed bug in  "Is Winner"

Из-за if, else if ... конструкции программа проверяла не все варианты выйграша, а стопалась на
else if (true) {if (false)}

kshrnkrstn committed:
Сhanging literals to constants
